rustls/src/crypto/aws_lc_rs/hpke.rs

```diff
@@ -0,0 +1,796 @@
+use alloc::boxed::Box;
+use alloc::sync::Arc;
+use alloc::vec::Vec;
+use core::fmt::{self, Debug, Formatter};
+
+use aws_lc_rs::aead::{
+    self, Aad, BoundKey, Nonce as AwsNonce, NonceSequence, OpeningKey, SealingKey, UnboundKey,
+    NONCE_LEN,
+};
+use aws_lc_rs::agreement;
+use aws_lc_rs::cipher::{AES_128_KEY_LEN, AES_256_KEY_LEN};
+use aws_lc_rs::digest::{SHA256_OUTPUT_LEN, SHA384_OUTPUT_LEN, SHA512_OUTPUT_LEN};
+use zeroize::Zeroize;
+
+use crate::crypto::aws_lc_rs::tls13::{
+    RING_HKDF_HMAC_SHA256, RING_HKDF_HMAC_SHA384, RING_HKDF_HMAC_SHA512,
+};
+use crate::crypto::hpke::{
+    EncapsulatedSecret, Hpke, HpkeOpener, HpkePrivateKey, HpkePublicKey, HpkeSealer, HpkeSuite,
+    HpkeSuites,
+};
+use crate::crypto::tls13::{expand, Hkdf, HkdfExpander};
+use crate::msgs::enums::{HpkeAead, HpkeKdf, HpkeKem};
+use crate::msgs::handshake::HpkeSymmetricCipherSuite;
+use crate::{Error, OtherError};
+
+/// Default [RFC 9180] Hybrid Public Key Encryption (HPKE) suites supported by aws-lc-rs cryptography.
+pub static ALL_SUPPORTED_SUITES: &HpkeSuites = &HpkeSuites(&[
+    DHKEM_P256_HKDF_SHA256_AES_128,
+    DHKEM_P256_HKDF_SHA256_AES_256,
+    DHKEM_P256_HKDF_SHA256_CHACHA20_POLY1305,
+    DHKEM_P384_HKDF_SHA384_AES_128,
+    DHKEM_P384_HKDF_SHA384_AES_256,
+    DHKEM_P384_HKDF_SHA384_CHACHA20_POLY1305,
+    DHKEM_P521_HKDF_SHA512_AES_128,
+    DHKEM_P521_HKDF_SHA512_AES_256,
+    DHKEM_P521_HKDF_SHA512_CHACHA20_POLY1305,
+    DHKEM_X25519_HKDF_SHA256_AES_128,
+    DHKEM_X25519_HKDF_SHA256_AES_256,
+    DHKEM_X25519_HKDF_SHA256_CHACHA20_POLY1305,
+]);
+
+/// Suite holds the concrete instantiations of the algorithms selected by the suite ID.
+#[derive(Clone)]
```

Comment:
Note to self: this doesn't need to be `Clone` - I think this was a leftover from some earlier work that fell away.